File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2020s/IZ08IO1/BridgeTest/src/FigureBridgeListbox.java
import javax.swing.DefaultListModel;
import javax.swing.JList;

public class FigureBridgeListbox extends FigureBridge
{
   private JList lb;
   public FigureBridgeListbox(JList lb)
   {
      this.lb=lb;
   }
   private DefaultListModel<Figure> getModel()
   {
      return (DefaultListModel<Figure>)lb.getModel();
   }
   @Override protected void initialize()
   {
      getModel().clear();
   }
   @Override protected void finalize()
   {
      lb.setSelectedIndex(getModel().size()-1);
   }
   @Override protected void proceed(Figure fig)
   {
      getModel().addElement(fig);
   }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2020/ID06TC1/BridgeTest/BridgeTest/FigureBridgeListbox.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace BridgeTest
{
    public class FigureBridgeListbox : FigureBridge
    {
        private ListBox listbox;
        public FigureBridgeListbox(ListBox listbox)
        {
            this.listbox = listbox;
        }
        protected override void initialize()
        {
            listbox.Items.Clear();
        }
        protected override void finalize()
        {
            listbox.SelectedIndex = listbox.Items.Count - 1;
        }
        protected override void proceed(Figure fig)
        {
            listbox.Items.Add(fig);
        }
    }
}
